include/gadget/macros.h

```diff
@@ -24,4 +24,23 @@
 	const void *gadget_snapshotter_##name##___##type __attribute__((unused)); \
 	const struct type *unusedevent_##name##___##type __attribute__((unused));
 
+#ifndef PROFILER_MAX_SLOTS
+#define PROFILER_MAX_SLOTS 27
+#endif /* !PROFILER_MAX_SLOTS */
+
+// hist_value is used as value for profiler hash map.
+struct hist_value {
+	__u32 slots[PROFILER_MAX_SLOTS];
+};
+
+const struct hist_value *unused___hist_value __attribute__((unused));
+
+// GADGET_PROFILER is used to define a profiler:
+// - name is the tracer name.
+// - hist_name is the name of the hash map used to send events to user space.
+// - hist_key_type is the type of map key.
+#define GADGET_PROFILER(name, hist_name, hist_key_type) \
+	const void *gadget_profiler_##name##___##hist_name##___##hist_key_type##___hist_value __attribute__((unused)); \
+	const struct hist_key_type *unusedevent_##name##___##hist_key_type __attribute__((unused));
```

Comment:
I think GADGET_TRACER, GADGET_PARAM, GADGET_SNAPSHOTTER, GADGET_PROFILER should be documented in the gadget helper API reference documentation:
https://github.com/inspektor-gadget/inspektor-gadget/blob/main/docs/reference/gadget-helper-api.md

Since this PR only introduce GADGET_PROFILER, the documentation could be added in a separate PR.

---

In the hello world gadget guide, we only mention `GADGET_TRACER()`:
https://github.com/inspektor-gadget/inspektor-gadget/blob/main/docs/devel/hello-world-gadget.md
Even if it's out of scope of this guide to explain GADGET_SNAPSHOTTER and GADGET_PROFILER, it could still have one sentence to say there are other kinds of gadgets and give a link to the reference documentation.